protocol/src/main/kotlin/models/Message.kt

```diff
@@ -118,6 +97,34 @@ sealed class Message {
         MessageKind.close -> objectMapper.readValue<Close>(payload)
       }
     }
+    
+    /**
+     * Verifies the cryptographic integrity of the message's signature.
+     *
+     * @throws Exception TODO link to crypto method throws
+     */
+    fun verify() {
+      // TODO detached payload sig check (regenerate payload and then check)
+    }
+
+    /**
+     * Validates a JSON payload based on tbDEX message JSON schemas.
+     *
+     * @param payload The JSON payload to validate.
+     * @throws Exception if the payload does not conform to the expected structure or data schema.
+     */
+    fun validate(payload: String) {
+      val messageJson = JSONObject(payload)
+
+      // validate message structure
+      Validator.validate(messageJson, "message")
+
+      val dataJson = messageJson.getJSONObject("data")
+      val kind = messageJson.getJSONObject("metadata").getString("kind")
+
+      // validate specific message data (Rfq, Quote, etc)
+      Validator.validate(dataJson, kind)
+    }
```

Comment:
potentially make these private functions?

Reply:
we use these in Message.validate() in MessageTest.kt... maybe not good to keep a method public purely for the sake of testing tho